gatherers/manifest.js

```diff
@@ -54,34 +32,62 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * Creates a listener for manifests in network requests, a load timeout, and
+   * a promise they resolve when complete.
+   * @private
+   */
+  static _resolveOnManifest() {
```

Comment:
I'm not sure if there's a great way to fundamentally rearrange this, though it can be cleaned up. Even if the two conditioned are `race`d, the load callback still needs a reference to `resolve` and we still need the callback and the promise in our load chain. I could maybe make it more explicit.